test/test.js

```diff
@@ -17,12 +24,47 @@ describe('primus-rooms-adapter', function () {
     expect(adapter.wildcard).to.be.an('object');
   });
 
-  describe('Adapter#del', function () {
+  describe.skip('Adapter#del', function () {
     it('should not leak memory', function () {
-      var adapter = new Adapter();
       adapter.del('foo');
       expect(adapter.rooms).to.eql({});
       expect(adapter.sids).to.eql({});
     });
   });
+  
+  describe('Adapter#add', function() {
+    it('should join a new room', function(done) {
```

Comment:
Mocha has built-in support for promises so we can remove `done` here and the `catch` from the chain below.